iam.tf

```
@@ -1,33 +1,49 @@
 locals {
-  iam_map = { for iam in var.iam : iam.role => iam }
+  # filter all objects that define a single role
+  iam_role = [for iam in var.iam : iam if can(iam.role)]
+
+  # filter all objects that define multiple roles and expand them to single roles
+  iam_roles = flatten([for iam in var.iam :
+    [for role in iam.roles : merge(iam, { role = role })] if can(iam.roles)
+  ])
+
+  # we allow to define role and roles in the same object and concatenate them
+  # each role can only be sepcified exactly once
+  iam = concat(local.iam_role, local.iam_roles)
+
+  iam_map = { for idx, iam in local.iam :
+    try(iam._key, iam.role) => idx
+  }
 }
 
 module "subscription-iam" {
-  source = "github.com/mineiros-io/terraform-google-pubsub-subscription-iam?ref=v0.0.3"
+  source = "github.com/mineiros-io/terraform-google-pubsub-subscription-iam?ref=v0.0.4"
 
   for_each = var.policy_bindings == null ? local.iam_map : {}
 
   module_enabled    = var.module_enabled
   module_depends_on = var.module_depends_on
 
-  subscription  = try(google_pubsub_subscription.subscription[0].name, null)
-  role          = each.value.role
-  members       = each.value.members
-  authoritative = try(each.value.authoritative, true)
+  subscription         = try(google_pubsub_subscription.subscription[0].name, null)
+  role                 = local.iam[each.value].role
+  members              = local.iam[each.value].members
+  computed_members_map = merge(var.computed_members_map, try(local.iam[each.value].computed_members_map, {}))
+  authoritative        = try(each.value.authoritative, true)
```

`each.value` is the index in the original list here. So this will always evaluate to `true`.

```suggestion
  authoritative        = try(local.iam[each.value].authoritative, true)
```